xarray_beam/_src/core.py

```diff
@@ -340,12 +340,14 @@ def _validate(self, dataset, split_vars):
         raise ValueError("dataset list cannot be empty")
     sizes = [ds.sizes for ds in self._datasets]
     if len({tuple(s.items()) for s in sizes}) > 1:
-      raise ValueError(f"inconsistent dataset sizes: {sizes}")
+      if any(len(sizes[0]) < len(s) for s in sizes[1:]):
+        raise ValueError(f"inconsistent dataset sizes, only the first can be greater: {sizes}")
     dv_shapes = [{k: v.shape for k, v in ds.items()} for ds in self._datasets]
     if split_vars and len({tuple(shape) for shape in dv_shapes}) > 1:
-      raise ValueError(
-          f"inconsistent data_var shapes when splitting variables: {dv_shapes}"
-      )
+      if any(len(dv_shapes[0].keys()) < len(dvs.keys()) for dvs in dv_shapes[1:]):
+        raise ValueError(
+            f"inconsistent data_var shapes when splitting variables: {dv_shapes}"
+        )
     chunks = [ds.chunks for ds in self._datasets]
     if len({tuple(c.items()) for c in chunks}) > 1:
       raise ValueError(f"inconsistent chunks: {chunks}")
```

Comment:
I think this check could probably be removed. Chunking with Dask is not directly tied to chunking when reading data in Xarray-Beam, and it could be valid to have different Dask chunk sizes on different datasets, especially if the Dask chunks always fit into Beam chunks.